src/D2L.CodeStyle.Analyzers/Async/Generator/FileCollector.cs

```diff
@@ -0,0 +1,273 @@
+﻿using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace D2L.CodeStyle.Analyzers.Async.Generator;
+
+internal partial class SyncGenerator {
+	/// <summary>
+	/// Creates a source file containing generated methods and the necessary
+	/// namespaces, using directives, and type declarations for them to live in
+	/// (but only the necessary ones).
+	/// </summary>
+	internal sealed class FileCollector {

```

Comment:
This piece isn't used yet, but there are some tests below that show what it's for.
The `Create()` API is a bit awkward to call, but it works well in the generator pipeline.

Reply:
Like at the point we use this we will have (possibly cached) generated source strings to replace select methods within a single file, and this things responsibility is to concatenate them with all the required usings/hierarchy (partial types/namespaces)